### DISCUSSION TITLE
Difference of Pooling Local vs Global

### DISCUSSION THREAD

**Discussion by joaovitorfranco** [alan-turing-institute/sktime#4747] [Q&A]
I've been exploring and trying to understand the different methods of configuring pooling for `make_reduction`. 

What I'm unable to understand is why with `pooling =  "local"` the model creates lags for `y` and `X` while with `pooling =  "global"` it creates lags only for `y`. 

I do understand that with local it will create a model for each instance and with global it will create only one model for all instances.

So my two main questions are:

1. Why do these methods have that behavior?
2. How is the format and how should I pass the `X_test` during the prediction step for each pooling method?

The code below can be used to replicate and show the difference in the number of features of each approach:

```python
import pandas as pd
import numpy as np

from sklearn.ensemble import RandomForestRegressor
from sktime.datasets import load_macroeconomic
from sktime.forecasting.compose import make_reduction


data = load_macroeconomic()
y = data["unemp"]
X = data.drop(columns=["unemp"])

regressor = RandomForestRegressor(random_state=42)

forecaster1 = make_reduction(
    regressor,
    window_length=10,
    strategy="recursive",
    pooling="local",
)

forecaster2 = make_reduction(
    regressor,
    window_length=10,
    strategy="recursive",
    pooling="global",
)

forecaster1.fit(X=X, y=y, fh=[1,2,3,4,5])
forecaster2.fit(X=X, y=y, fh=[1,2,3,4,5])
print(f'Number of features with pooling local: {forecaster1.get_fitted_params()["estimator__n_features_in"]}')
print(f'Number of features with pooling global: {forecaster2.get_fitted_params()["estimator__n_features_in"]}')
```

**Comment by joaovitorfranco** [comment]
Does anyone have an understanding of these configurations and behaviors?

**Reply by fkiraly** [reply]
apologies, I did not spot this.

What you are reporting actually sounds like a bug, since the behaviour for `X` seems inconsistent between the local/global settings.

FYI @davidgilbertson, @danbartl - do you have any further insight perhaps?

In general, `X_test` should be in exactly the same format as `X` - for instance, a `pd.DataFrame` with 2 level `pd.MultiIndex`.

**Reply by fkiraly** [reply]
In case we confirm this is different between the settings, I would consider this a bug and move this to the issue tracker.

**Reply by joaovitorfranco** [reply]
So should we transfer it to the bug section? I was kind of confused at first, but if this behavior is not intended then makes sense to address it as a bug.

**Reply by fkiraly** [reply]
yes, let's move it to bugs.

**Reply by fkiraly** [reply]
moved here: https://github.com/sktime/sktime/issues/4847
